### PR TITLE
Create LoadBalancer in nginx ingress tests

### LINKED ISSUE #77538: Some [sig-network] tests don't work on "private" clusters
Hello sig-network,

We, sig-scalability, are exploring using "private" clusters (i.e. the ones where nodes don't have public IPs) for our large (up to 5K nodes) scalability tests to avoid consuming expensive public IPs. More details in https://github.com/kubernetes/kubernetes/issues/76374

With a few minor modifications we managed to get an almost passing correctness test suite on a private cluster except the 3 sig-network tests which explicitly assume that nodes are reachable over public IPs. 

**Failing tests** ([testgrid link](https://k8s-testgrid.appspot.com/sig-scalability-experiments#gce-private-cluster-correctness)):
* [sig-network] Firewall rule should have correct firewall rules for e2e cluster
* [sig-network] Loadbalancing: L7 [Slow] Nginx should conform to Ingress spec
* [sig-network] Services should be able to create a functioning NodePort service

Could you take a look at them and advise if it would be possible to rewrite them so they can work both in the "public" and "private" clusters? 

We'd really appreciate your help assistance here.
Let me know if you have any questions.

Thanks,

Matt

### PR DESCRIPTION
**Pull request by mm4tt**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation

/kind failing-test

> /kind feature
> /kind flake

**What this PR does / why we need it**:
It fixes the nginx ingress tests that is currently failing in the private clusters where nodes don't have public ips, see https://k8s-testgrid.appspot.com/sig-scalability-experiments#gce-private-cluster-correctness.
It achieves that by creating a load balancer in front of the nginx controller and using the [`--publish-service` nginx ingress feature](https://github.com/kubernetes/ingress-nginx/tree/master/docs/examples/static-ip)

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubernetes/kubernetes/issues/77538

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/framework/ingress/ingress_utils.go [hunk 1]
```diff
@@ -831,11 +831,23 @@ type NginxIngressController struct {
 	rc         *v1.ReplicationController
 	pod        *v1.Pod
 	Client     clientset.Interface
-	externalIP string
 }
 
 // Init initializes the NginxIngressController
 func (cont *NginxIngressController) Init() {
+	framework.Logf("Creating load balancer service for nginx ingress controller")
```
Comment (wojtek-t) [inline test/e2e/framework/ingress/ingress_utils.go]: Add a comment why you do that (i.e. to make this work also on clusters in which nodes doesn't have public IPs).

Comment (mm4tt) [reply test/e2e/framework/ingress/ingress_utils.go]: Done. Added a comment.

**Comment by wojtek-t** [comment]
LGTM but I will let @freehan to also take a look

**Review by freehan — COMMENTED** [review]
you might want to filter it based on cloud provider? I would assume this e2e may run in other environment and there is no LB type service.

**Comment by mm4tt** [comment]
> you might want to filter it based on cloud provider? I would assume this e2e may run in other environment and there is no LB type service.

Looks like this test is already skipped if provider is not gce|gke - https://github.com/kubernetes/kubernetes/blob/7ca57877b00aa91f0a240a508c38cbd37a75066c/test/e2e/network/ingress.go#L717

I think we should be good, but please correct me if I'm wrong.

**Pull request merged**